packages/admin/admin/src/form/FinalFormChooseFile.tsx

```diff
@@ -0,0 +1,269 @@
+            float: "right",
+            position: "absolute",
+            top: 0,
+            right: 0,
+            padding: "10px",
+            color: palette.error.main,
+        },
+        fileList: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            gap: "2px",
+            width: "100%",
+        },
+        fileListItem: {
+            display: "flex",
+            padding: "8px 7px 8px 15px",
+            alignItems: "center",
+            gap: "10px",
+            borderRadius: "4px",
+            background: palette.background.default,
+            justifyContent: "space-between",
+            minWidth: "250px",
+            width: "100%",
+            boxSizing: "border-box",
+        },
+        fileListItemInfos: {
+            display: "flex",
+            justifyContent: "end",
+            gap: "10px",
+        },
+        rejectedFileListItem: {
+            display: "flex",
+            padding: "8px 7px 8px 15px",
+            alignItems: "center",
+            gap: "10px",
+            borderRadius: "4px",
+            background: palette.background.default,
+            justifyContent: "space-between",
+            minWidth: "250px",
+            border: `1px dashed ${palette.error.main}`,
+            color: palette.error.main,
+            width: "100%",
+            boxSizing: "border-box",
+        },
+        errorMessage: {
+            display: "flex",
+            alignItems: "center",
+            color: palette.error.main,
+            gap: "5px",
+        },
+        droppableAreaIsDisabled: {},
+        droppableAreaHasError: {},
+        fileListText: {
+            textOverflow: "ellipsis",
+            overflow: "hidden",
+            whiteSpace: "nowrap",
+        },
+    });
+};
+
+export interface FinalFormChooseFileProps extends FieldRenderProps<File | File[], HTMLInputElement> {
+    disableDropzone?: boolean;
+    disableSelectFileButton?: boolean;
+    accept: Accept;
+    maxSize: number;
+    maxFiles?: number;
+    iconMapping?: {
+        delete?: React.ReactNode;
+        info?: React.ReactNode;
+        select?: React.ReactNode;
+    };
+}
+
+const FinalFormChooseFileComponent: React.FunctionComponent<WithStyles<typeof styles> & FinalFormChooseFileProps> = ({
+    classes,
+    disabled,
+    disableDropzone,
+    disableSelectFileButton,
+    accept,
+    maxSize = 50 * 1024 * 1024,
+    maxFiles,
+    input: { onChange, value: fieldValue, multiple: multipleFiles },
+    iconMapping = {},
+}) => {
+    const { delete: deleteIcon = <Delete />, info: infoIcon = <Info />, select: selectIcon = <Select /> } = iconMapping;
+
+    // For multipleFiles, if maxFiles is not set, or if maxFiles is set and the number of the files in the current fieldValue together with the acceptedFiles is equal or less then maxFiles, add acceptedFiles to the current fieldValue. Else replace the fieldValue with acceptedFiles
```

Comment:
I don't think this comment is necessary as it just describes the `if`s that I can read

```suggestion
```